e2e/restart_test.go

```diff
@@ -20,23 +17,21 @@ func TestRestart(t *testing.T) {
 	r.Nil(err)
 	r.True(gd.IsRunning())
 
-	reqT := `
-{
-  "name": "vol1",
-  "bricks": ["%s:%s"],
-  "force": true
-}
-	`
 	dir, err := ioutil.TempDir("", "")
 	r.Nil(err)
 	defer os.RemoveAll(dir)
 
-	req := fmt.Sprintf(reqT, gd.PeerAddress, dir)
+	client := initRestclient(gd.ClientAddress)
 
-	resp, err := http.Post(fmt.Sprintf("http://%s/v1/volumes", gd.ClientAddress), "application/json", strings.NewReader(req))
-	r.Nil(err)
-	defer resp.Body.Close()
-	r.Equal(http.StatusCreated, resp.StatusCode)
+	createReq := api.VolCreateReq{
+		Name: "vol1",
+		Bricks: []string{
+			gd.PeerAddress + ":" + dir,
+		},
+		Force: true,
+	}
+	_, errVolCreate := client.VolumeCreate(createReq)
+	r.Nil(errVolCreate)
```

Comment:
Behavior is very confusing in Go :(

Do you have any doc link which explains this better?

    resp, err := SomeFunc()
    resp, err := SomeOtherFunc()  # Not valid
    resp, err1 := SomeOtherFunc()  # Valid since one new variable
    resp, err = SomeOtherFunc()  # Valid
    resp1, err = SomeOtherFunc() # Not Valid, resp1 is new variable